File: src-tauri/src/models/background_process.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum BackgroundProcessStatus {
    Queued,
    Running,
    Paused,
    Completed,
    Failed,
    Cancelled,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct BackgroundProcess {
    pub process_id: String,
    pub process_type: String,
    pub status: BackgroundProcessStatus,
    pub priority: u32,
    pub total_items: u64,
    pub processed_items: u64,
    pub failed_items: u64,
    pub remark: Option<String>,
    pub created_at_ms: u64,
    pub updated_at_ms: u64,
    pub started_at_ms: Option<u64>,
    pub finished_at_ms: Option<u64>,
}

File: src/features/import/types/background-process.ts
export type BackgroundProcessStatus =
  | "queued"
  | "running"
  | "paused"
  | "completed"
  | "failed"
  | "cancelled";

export interface BackgroundProcess {
  processId: string;
  processType: string;
  status: BackgroundProcessStatus;
  priority: number;
  totalItems: number;
  processedItems: number;
  failedItems: number;
  remark: string | null;
  createdAtMs: number;
  updatedAtMs: number;
  startedAtMs: number | null;
  finishedAtMs: number | null;
}
